packages/components/src/XComponent.tsx

```diff
@@ -53,13 +53,17 @@ export function createXComponent<P extends Record<string, unknown>>(
     );
   }
 
+  // Add the display name so this appears as a tag in the React DevTools
+  // Need to add it here, and then when it's wrapped with the `forwardRef` it will automatically get the display name of the original component
+  XComponent.displayName = `XComponent(${
+    Component.displayName ?? Component.name
+  })`;
```

Comment:
Very minor, but do we want to change the display name after `forwardRef` instead? The benefit would just be in dev tools we'd see `XComponent + 1` instead of `ForwardRef + 1`. Other than that I think they'd be identical